view/frontend/web/js/components/compare-products.js

```diff
@@ -1,8 +1,47 @@
 (function () {
-    'use strict';
+    "use strict";
 
-    $.view('compareProducts', {
-        component: 'Magento_Catalog/js/view/compare-products',
-        compareProducts: $.customerData.get('compare-products')
+    $.view("compareProducts", {
+        component: "Magento_Catalog/js/view/compare-products",
+        compareProducts: $.customerData.get("compare-products"),
+    });
+
+    $.widget('mage.compareList', {
+        component: 'compareList',
+
+        /** @inheritdoc */
+        _create: function () {
+            var elem = this.element,
+                products = $('thead td', elem),
+                headings;
+
+            if (products.length > this.options.productsInRow) {
```

Comment:
Please remove everything from the `_create` function except print functionality.

Reply:
@vovayatsyuk done